src/HIE/Bios/Environment.hs

```diff
@@ -53,13 +56,22 @@ initSession  ComponentOptions {..} = do
 
 ----------------------------------------------------------------
 
--- | Obtain the directory for system libraries.
-getSystemLibDir :: IO (Maybe FilePath)
-getSystemLibDir = do
-    res <- readProcess "ghc" ["--print-libdir"] []
-    return $ case res of
-        ""   -> Nothing
-        dirn -> Just (init dirn)
+-- | Obtain the directory for the ghc library directory.
+-- It tries this specific order of paths:
+-- 1. the NIX_GHC_LIBDIR if it is set
+-- 2. calling --print-libdir with the cradle's ghc
+-- 3. using ghc-paths
+-- Note that 3 will only work if we are running on the same machine that
+-- hie-bios compiled on, so we avoid using it unless we have to!
+getGhcLibDir :: Cradle a -> IO FilePath
+getGhcLibDir cradle =
+  fmap (fromMaybe Paths.libdir) $ runMaybeT $ fromNix <|> fromCradle
```

Comment:
Do we know what should take precedence? 
In my experience with nix, ghc is wrapped to give the correct libdir.
The order should be in my opinion: Cradle Ghc -> Nix env var -> Paths.libdir

Reply:
As ghc-paths will be wrong in the general case maybe we should remove it and returns `IO (Maybe Filepath)` to make clear we cant return it for all cases in a reliable way and force the client code handle that possibility (maybe using ghc-paths at the end!)

Reply:
My aim here was to mirror what currently happens inside ghcide/hls, and I presume this nix env var might be used for the direct cradles where it's calling `ghc` without cabal or stack: https://github.com/digital-asset/ghcide/blob/9e984b56cfb62a83829975210ce3f14c668912f6/docs/Setup.md#issues-with-nix

Reply:
If the client code is just going to end up calling ghc-paths then shouldn't we just do it here as well? We might want to rename `getGhcLibDir` to something more honest though like `mostlyAccurateGuessGhcLibDir`, since the *only* scenario where ghc-paths doesn't work will be when moving binaries across machines. On the other hand, if we are using static binaries then we would really want to error here instead of silently giving the wrong path. But then we would end up splitting the libdir logic even further back into the clients

Reply:
Well i suppose client code could do a final sanity check executing the ghc returned by the cradle (`--numeric-version` ?) and catch the error. Not elegant but doable

Reply:
@jneira `getGhcOnPathLibDir` should now do a check of the ghc version beforehand now